streaming/src/queue/transport.cc

```diff
@@ -0,0 +1,116 @@
+#include "transport.h"
+#include "utils.h"
+
+namespace ray {
+namespace streaming {
+
+static constexpr int TASK_OPTION_RETURN_NUM_0 = 0;
+static constexpr int TASK_OPTION_RETURN_NUM_1 = 1;
+
+void Transport::Send(std::unique_ptr<LocalMemoryBuffer> buffer, RayFunction &function) {
+  STREAMING_LOG(INFO) << "Transport::Send buffer size: " << buffer->Size();
+  std::unordered_map<std::string, double> resources;
+  TaskOptions options{TASK_OPTION_RETURN_NUM_0, true, resources};
+
+  char meta_data[3] = {'R', 'A', 'W'};
+  std::shared_ptr<LocalMemoryBuffer> meta =
+      std::make_shared<LocalMemoryBuffer>((uint8_t *)meta_data, 3, true);
+
+  std::vector<TaskArg> args;
+  if (function.GetLanguage() == Language::PYTHON) {
+    auto dummy = "__RAY_DUMMY__";
+    std::shared_ptr<LocalMemoryBuffer> dummyBuffer =
+        std::make_shared<LocalMemoryBuffer>((uint8_t *)dummy, 13, true);
+    args.emplace_back(TaskArg::PassByValue(
+        std::make_shared<RayObject>(std::move(dummyBuffer), meta, true)));
+  }
+  args.emplace_back(
+      TaskArg::PassByValue(std::make_shared<RayObject>(std::move(buffer), meta, true)));
+
+  STREAMING_CHECK(core_worker_ != nullptr);
+  std::vector<ObjectID> return_ids;
+  std::vector<std::shared_ptr<RayObject>> results;
+  ray::Status st =
+      core_worker_->SubmitActorTask(peer_actor_id_, function, args, options, &return_ids);
+  if (!st.ok()) {
+    STREAMING_LOG(ERROR) << "SubmitActorTask failed. " << st;
+  }
+
+  Status get_st = core_worker_->Get(return_ids, -1, &results);
+  if (!get_st.ok()) {
+    STREAMING_LOG(ERROR) << "Get fail.";
+  }
+}
+
+std::shared_ptr<LocalMemoryBuffer> Transport::SendForResult(
```

Comment:
this function has duplicated code with the above one, can we unify them?

Reply:
> this function has duplicated code with the above one, can we unify them?

fixed.